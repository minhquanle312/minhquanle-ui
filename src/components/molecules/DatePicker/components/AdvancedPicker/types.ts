export type TOption = {
  dateType: TDateType;
  calculationType: TCalculationType;
  calculationDate: TCalculationDate;
  value: number;
};

export type TShowCalculationTypeCondition = {
  dateType: Partial<Record<TDateType, TCalculationType[]>>;
};

export type TDateType =
  | 'today'
  | 'first_day_of_week_mon_sun'
  | 'first_day_of_week_sun_sat'
  | 'first_day_of_month'
  | 'first_day_of_quarter'
  | 'first_day_of_year'
  | 'last_day_of_week_mon_sun'
  | 'last_day_of_week_sun_sat'
  | 'last_day_of_month'
  | 'last_day_of_quarter'
  | 'last_day_of_year'
  | 'fixed';

export type TCalculationType = 'minus' | 'plus' | '';

export type TCalculationDate =
  | 'seconds'
  | 'minutes'
  | 'hours'
  | 'days'
  | 'weeks'
  | 'months'
  | 'quarters'
  | 'years'
  | '';

export type TOperatorKey = 'after' | 'between' | 'before';

export type TAdvancedType = 'startDate' | 'endDate';
