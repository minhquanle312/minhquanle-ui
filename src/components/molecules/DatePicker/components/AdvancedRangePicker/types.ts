// Types
import { TCalculationType, TOption } from '../AdvancedPicker/types';

export type TOnChangeMode = 'system' | 'user';

export type TTimeRange = {
  startDate: TOption & { date: string };
  endDate: TOption & { date: string };
};

export type TOnChangePayload = {
  timeRange: TTimeRange;
  mode: 'system' | 'user';
};

export type TDateConfig = {
  calculationTypeKeysShow?: TCalculationType[];
};
