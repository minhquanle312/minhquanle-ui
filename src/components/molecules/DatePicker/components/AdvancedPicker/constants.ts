// Types
import { TAdvancedType } from './types';

export const DATE_TYPES = [
  { value: 'today', label: 'datePicker.dateTypes.today' },
  { value: 'first_day_of_week_mon_sun', label: 'datePicker.dateTypes.firstDayOfWeekMonSun' },
  { value: 'first_day_of_week_sun_sat', label: 'datePicker.dateTypes.firstDayOfWeekSunSat' },
  { value: 'first_day_of_month', label: 'datePicker.dateTypes.firstDayOfMonth' },
  { value: 'first_day_of_quarter', label: 'datePicker.dateTypes.firstDayOfQuarter' },
  { value: 'first_day_of_year', label: 'datePicker.dateTypes.firstDayOfYear' },
  { value: 'last_day_of_week_mon_sun', label: 'datePicker.dateTypes.lastDayOfWeekMonSun' },
  { value: 'last_day_of_week_sun_sat', label: 'datePicker.dateTypes.lastDayOfWeekSunSat' },
  { value: 'last_day_of_month', label: 'datePicker.dateTypes.lastDayOfMonth' },
  { value: 'last_day_of_quarter', label: 'datePicker.dateTypes.lastDayOfQuarter' },
  { value: 'last_day_of_year', label: 'datePicker.dateTypes.lastDayOfYear' },
  { value: 'fixed', label: 'datePicker.dateTypes.fixed' },
];

export const CALCULATION_TYPES = [
  { value: 'minus', label: 'global.minus' },
  { value: 'plus', label: 'global.plus' },
];

export const CALCULATION_DATES = [
  { value: 'seconds', label: 'datePicker.seconds' },
  { value: 'minutes', label: 'datePicker.minutes' },
  { value: 'hours', label: 'datePicker.hours' },
  { value: 'days', label: 'datePicker.days' },
  { value: 'weeks', label: 'datePicker.weeks' },
  { value: 'months', label: 'datePicker.months' },
  { value: 'quarters', label: 'datePicker.quarters' },
  { value: 'years', label: 'datePicker.years' },
];

export const VALUE_TYPES = {
  INPUT_TEXT: 'INPUT_TEXT',
  INPUT_NUMBER: 'INPUT_NUMBER',
  MATCHES_ANY: 'MATCHES_ANY',
  YEAR: 'YEAR',
  YEAR_QUARTER: 'YEAR_QUARTER',
  YEAR_MONTH: 'YEAR_MONTH',
  YEAR_WEEK: 'YEAR_WEEK',
  YEAR_MONTH_DAY: 'YEAR_MONTH_DAY',
  YEAR_MONTH_DAY_HOUR: 'YEAR_MONTH_DAY_HOUR',
  YEAR_MONTH_DAY_MINUTE: 'YEAR_MONTH_DAY_MINUTE',
  YEAR_MONTH_DAY_SECOND: 'YEAR_MONTH_DAY_SECOND',
  QUARTER: 'QUARTER',
  MONTH: 'MONTH',
  WEEK: 'WEEK',
  MONTH_DAY: 'MONTH_DAY',
  DAY_OF_WEEK: 'DAY_OF_WEEK',
  DAY: 'DAY',
  HOUR: 'HOUR',
  MINUTE: 'MINUTE',
  INPUT_TEXT_NO_FIELD: 'INPUT_TEXT_NO_FIELD',
};

export const DEFAULT_DATE_FORMAT = 'YYYYMMDDHHmmss';
export const DEFAULT_TIME_FORMAT = 'HH:mm:ss';
export const DEFAULT_DATE_DISPLAY_FORMAT = 'DD/MM/YYYY HH:mm:ss';

export const TIME_PICKER_TYPE = {
  DATE_TIME: 'DateTime',
  DATE_HOUR: 'DateHour',
  DATE_HOUR_MINUTE: 'DateHourMinute',
  DATE: 'Date',
};

export const YEAR_PICKER_TYPE = {
  YEAR: 'year',
  MONTH: 'month',
  QUARTER: 'quarter',
};

export const ADVANCED_PICKER_TYPE: Record<string, { value: TAdvancedType; label: string }> = {
  START_DATE: {
    value: 'startDate',
    label: 'datePicker.startDate',
  },
  END_DATE: {
    value: 'endDate',
    label: 'datePicker.endDate',
  },
};
