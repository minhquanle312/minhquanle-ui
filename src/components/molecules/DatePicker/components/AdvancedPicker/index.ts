export { AdvancedPicker } from './AdvancedPicker';

export type { AdvancedPickerProps } from './AdvancedPicker';
export type { TOption as TAdvancedPickerOption } from './types';
