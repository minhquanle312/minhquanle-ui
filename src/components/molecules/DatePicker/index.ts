export { DatePicker } from './DatePicker';

export type { AdvancedPickerProps, TAdvancedPickerOption } from './components/AdvancedPicker';
export type { TAdvancedRangePickerTimeRange } from './components/AdvancedRangePicker';
