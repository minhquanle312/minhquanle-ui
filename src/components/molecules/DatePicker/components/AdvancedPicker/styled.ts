// Libraries
import { THEME } from '../../../../../constants';
import styled from 'styled-components';

export const DropdownContent = styled.div`
  width: 230px;
  padding: 10px 10px 0px 10px;
  display: flex;
  flex-direction: column;
  gap: 10px;
`;

export const DropdownHeader = styled.div`
  padding: 10px 10px 0px 10px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  text-align: center;

  > .__title {
    color: ${THEME.token?.bw6};
  }
`;

export const DropdownFooter = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 0 10px 10px 10px;
`;

export const DatePickerHeader = styled.div`
  padding: 10px 10px 0px 10px;
  display: flex;
  flex-direction: column;
  gap: 10px;
  line-height: 1.6 !important;
`;

export const DatePickerCustomInput = styled.input`
  font-family: ${THEME.token?.fontFamily};
  color: ${THEME.token?.bw10} !important;
`;

export const DatePickerFooter = styled.div`
  position: absolute;
  bottom: 0px;
  width: 100%;
`;

export const CalendarIconWrapper = styled.div`
  display: inline-flex;
  cursor: pointer;
  pointer-events: all;
`;
