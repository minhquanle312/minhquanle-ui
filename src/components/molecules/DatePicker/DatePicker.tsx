// Libraries
import { DatePicker as AntdDatePicker } from 'antd';

// Components
import { AdvancedPicker } from './components/AdvancedPicker';
import { AdvancedRangePicker } from './components/AdvancedRangePicker';

export type TDatePicker = typeof AntdDatePicker & {
  AdvancedPicker: typeof AdvancedPicker;
  AdvancedRangePicker: typeof AdvancedRangePicker;
};

export const DatePicker = AntdDatePicker as TDatePicker;

DatePicker.AdvancedPicker = AdvancedPicker;
DatePicker.AdvancedRangePicker = AdvancedRangePicker;
