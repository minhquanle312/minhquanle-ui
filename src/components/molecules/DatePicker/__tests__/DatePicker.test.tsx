// Libraries
import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import dayjs from 'dayjs';

// Components
import { AdvancedPickerProps, DatePicker } from '..';
import { Default } from '../DatePicker.stories';
import { AdvancedPicker } from '../components/AdvancedPicker';
import { Button } from '../../../atoms';
import { Dropdown } from '../..';

// Constants
import {
  ADVANCED_PICKER_TYPE,
  DEFAULT_DATE_FORMAT,
  VALUE_TYPES,
} from '../components/AdvancedPicker/constants';

// Types
import { TOperatorKey } from '../components/AdvancedPicker/types';

// Styleds
import { DatePickerCustomInput } from '../components/AdvancedPicker/styled';

// Mock dependencies
jest.mock('react-i18next', () => ({ useTranslation: () => ({ t: jest.fn() }) }));

describe('DatePicker Component', () => {
  describe('Basic DatePicker Component', () => {
    it('should render the Basic DatePicker Component', () => {
      const { container } = render(<DatePicker />);

      expect(container).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });
  describe('Default DatePicker Component', () => {
    it('should render the Default DatePicker Component', () => {
      const args = {};
      const { container } = render(<Default {...args} />);

      expect(container).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });

  describe('DatePickerCustomInput Component', () => {
    it('should check onClick event of DatePickerCustomInput', () => {
      const toggleOpenDropdown = jest.fn();
      const { getByPlaceholderText } = render(
        <DatePickerCustomInput readOnly placeholder="Select Date" onClick={toggleOpenDropdown} />,
      );

      const input = getByPlaceholderText('Select Date');
      fireEvent.click(input);

      expect(toggleOpenDropdown).toBeCalledTimes(1);
    });
  });

  describe('Advanced DatePicker Component', () => {
    const args: AdvancedPickerProps = {
      label: '',
      type: ADVANCED_PICKER_TYPE.START_DATE.value,
      operatorKey: 'after',
      dateTypeKeysShow: ['today'],
      date: dayjs().format(DEFAULT_DATE_FORMAT),
      format: DEFAULT_DATE_FORMAT,
      inputStyle: {
        width: 120,
      },
      option: {
        dateType: 'today',
        calculationType: 'minus',
        calculationDate: 'seconds',
        value: 0,
      },
      calculationTypeKeysShow: ['minus', 'plus'],
      showCalculationTypeCondition: {
        dateType: {
          today: ['minus'],
        },
      },
      formatInputDisplay: 'MMM DD, YYYY',
      valueType: 'default',
      showTime: true,
      disabled: false,
    };

    it('should render the Advanced DatePicker Component', () => {
      const { container } = render(<AdvancedPicker {...args} />);

      expect(container).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });

    it('should render with the defaultDateTypeKey', () => {
      const args: AdvancedPickerProps = { defaultDateTypeKey: 'today' };
      const { container } = render(<AdvancedPicker {...args} />);

      expect(container).toBeInTheDocument();
    });

    it('should render without the defaultDateTypeKey', () => {
      const args: AdvancedPickerProps = {};
      const { container } = render(<AdvancedPicker {...args} />);

      expect(container).toBeInTheDocument();
    });

    it('should render newCalculationDates ', () => {
      const ValueTypes = [
        VALUE_TYPES.YEAR_MONTH_DAY_HOUR,
        VALUE_TYPES.YEAR_MONTH_DAY_MINUTE,
        VALUE_TYPES.YEAR,
        VALUE_TYPES.YEAR_MONTH,
        VALUE_TYPES.YEAR_MONTH_DAY_SECOND,
        VALUE_TYPES.YEAR_MONTH_DAY,
      ];

      ValueTypes.forEach(valueType => {
        const args: AdvancedPickerProps = { valueType };
        const { container } = render(<AdvancedPicker {...args} />);
        expect(container).toBeInTheDocument();
      });
    });

    it('should handle detect day is start or end', () => {
      const operatorKeys: TOperatorKey[] = ['before', 'between', 'after'];

      operatorKeys.forEach(operatorKey => {
        const args: AdvancedPickerProps = { operatorKey };
        const { container } = render(<AdvancedPicker {...args} />);

        expect(container).toBeInTheDocument();
      });
    });

    it('should show CalculationTypeCondition ', () => {
      const args: AdvancedPickerProps = {
        showCalculationTypeCondition: {
          dateType: {
            today: ['minus'],
          },
        },
      };

      const { container } = render(<AdvancedPicker {...args} />);
      expect(container).toBeInTheDocument();
    });

    it('should show CalculationTypeCondition with empty value', () => {
      const args: AdvancedPickerProps = {
        calculationTypeKeysShow: [''],
        showCalculationTypeCondition: {
          dateType: {
            fixed: [''],
          },
        },
      };

      const { container } = render(<AdvancedPicker {...args} />);
      expect(container).toBeInTheDocument();
    });

    it("should handle set state with newDateType.value === 'fixed'", () => {
      const args: AdvancedPickerProps = {
        option: {
          dateType: 'fixed',
          calculationType: 'minus',
          calculationDate: 'seconds',
          value: 0,
        },
      };

      const { container } = render(<AdvancedPicker {...args} />);
      expect(container).toBeInTheDocument();
    });

    it('should handle set state with invalid date', () => {
      const args: AdvancedPickerProps = {
        date: '2023/7-5',
        option: {
          dateType: 'fixed',
          calculationType: 'minus',
          calculationDate: 'seconds',
          value: 0,
        },
        showCalculationTypeCondition: {
          dateType: { first_day_of_year: ['plus'] },
        },
      };

      const { container } = render(<AdvancedPicker {...args} />);
      expect(container).toBeInTheDocument();
    });

    it('should handle click button Apply', () => {
      const onClickApply = jest.fn();
      const { getByText } = render(
        <Button type="primary" onClick={onClickApply}>
          Apply
        </Button>,
      );

      const button = getByText('Apply');

      fireEvent.click(button);
      expect(onClickApply).toBeCalled();
    });

    it('should render Text Component when label is not empty', () => {
      const args: AdvancedPickerProps = {
        label: 'text',
      };

      const { container } = render(<AdvancedPicker {...args} />);
      expect(container).toBeInTheDocument();
    });

    it('should handle inputRender when DatePicker is changed', () => {
      const handleInputRender = jest.fn();
      const { container } = render(
        <DatePicker
          disabled={false}
          allowClear={false}
          inputReadOnly
          inputRender={handleInputRender}
        />,
      );
      fireEvent.change(container);
      expect(handleInputRender).toHaveBeenCalledTimes(1);
      expect(container).toBeInTheDocument();
    });

    it('should handle onOpenChange when Dropdown is changed', () => {
      // Mock toggleOpenDropdown function
      const toggleOpenDropdown = jest.fn();

      const { getByRole, getByText } = render(
        <Dropdown
          disabled={false}
          open
          dropdownRender={() => <div>Dropdown Content</div>}
          trigger={['click']}
          onOpenChange={toggleOpenDropdown}
        >
          <input
            type="text"
            disabled={false}
            onClick={toggleOpenDropdown}
            readOnly
            value="Dropdown Input"
          />
        </Dropdown>,
      );

      // Find the input element
      const inputElement = getByRole('textbox');

      // Trigger click event on input element
      fireEvent.click(inputElement);

      // Check if toggleOpenDropdown function has been called
      expect(toggleOpenDropdown).toHaveBeenCalled();

      // Find the dropdown content
      const dropdownContent = getByText('Dropdown Content');

      // Check if the dropdown content is rendered
      expect(dropdownContent).toBeInTheDocument();
    });

    it('should render the Advanced DatePicker Component with dateTypeKeysShow = fixed', () => {
      const args: AdvancedPickerProps = {
        dateTypeKeysShow: ['fixed'],
      };

      const { container } = render(<AdvancedPicker {...args} />);

      expect(container).toBeInTheDocument();
    });

    it('should render the Advanced DatePicker Component with errorMessage', () => {
      const args: AdvancedPickerProps = {
        errorMessage: 'message',
      };

      const { container } = render(<AdvancedPicker {...args} />);

      expect(container).toBeInTheDocument();
    });

    it('should render the Advanced DatePicker Component with disabled', () => {
      const args: AdvancedPickerProps = {
        disabled: true,
      };

      const { container } = render(<AdvancedPicker {...args} />);

      expect(container).toBeInTheDocument();
    });
  });
});
