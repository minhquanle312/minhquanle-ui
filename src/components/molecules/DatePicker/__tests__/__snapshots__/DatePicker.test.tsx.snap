// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker Component Advanced DatePicker Component should render the Advanced DatePicker Component 1`] = `
<div>
  <div
    class="ant-space css-dev-only-do-not-override-htwhyh ant-space-vertical"
  >
    <div
      class="ant-space-item"
      style="margin-bottom: 5px;"
    />
    <div
      class="ant-space-item"
    >
      <span
        class="ant-input-affix-wrapper ant-input-affix-wrapper-readonly ant-dropdown-trigger css-dev-only-do-not-override-htwhyh"
        style="width: 120px;"
      >
        <input
          class="ant-input css-dev-only-do-not-override-htwhyh"
          readonly=""
          type="text"
          value="Jul 09, 2023"
        />
        <span
          class="ant-input-suffix"
        >
          <div
            class="sc-fsQiph kulHsB"
            style="pointer-events: all;"
          >
            <svg
              fill="currentColor"
              height="19"
              viewBox="0 0 24 24"
              width="19"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zm0-12H5V6h14v2zm-7 5h5v5h-5z"
              />
            </svg>
          </div>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Component Advanced Range Picker shoud render AdvancedRangePicker component with default props 1`] = `
<div>
  <div
    class="ant-space css-dev-only-do-not-override-htwhyh ant-space-horizontal ant-space-align-center"
  >
    <div
      class="ant-space-item"
      style="margin-right: 20px;"
    >
      <div
        class="ant-space css-dev-only-do-not-override-htwhyh ant-space-vertical"
      >
        <div
          class="ant-space-item"
          style="margin-bottom: 5px;"
        />
        <div
          class="ant-space-item"
        >
          <span
            class="ant-input-affix-wrapper ant-input-affix-wrapper-readonly ant-dropdown-trigger css-dev-only-do-not-override-htwhyh"
            style="width: 120px;"
          >
            <input
              class="ant-input css-dev-only-do-not-override-htwhyh"
              readonly=""
              type="text"
              value="Jul 09, 2022"
            />
            <span
              class="ant-input-suffix"
            >
              <div
                class="sc-fsQiph kulHsB"
                style="pointer-events: all;"
              >
                <svg
                  fill="currentColor"
                  height="19"
                  viewBox="0 0 24 24"
                  width="19"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zm0-12H5V6h14v2zm-7 5h5v5h-5z"
                  />
                </svg>
              </div>
            </span>
          </span>
        </div>
      </div>
    </div>
    <div
      class="ant-space-item"
    >
      <div
        class="ant-space css-dev-only-do-not-override-htwhyh ant-space-vertical"
      >
        <div
          class="ant-space-item"
          style="margin-bottom: 5px;"
        />
        <div
          class="ant-space-item"
        >
          <span
            class="ant-input-affix-wrapper ant-input-affix-wrapper-readonly ant-dropdown-trigger css-dev-only-do-not-override-htwhyh"
            style="width: 120px;"
          >
            <input
              class="ant-input css-dev-only-do-not-override-htwhyh"
              readonly=""
              type="text"
              value="Jul 08, 2023"
            />
            <span
              class="ant-input-suffix"
            >
              <div
                class="sc-fsQiph kulHsB"
                style="pointer-events: all;"
              >
                <svg
                  fill="currentColor"
                  height="19"
                  viewBox="0 0 24 24"
                  width="19"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M0 0h24v24H0V0z"
                    fill="none"
                  />
                  <path
                    d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zm0-12H5V6h14v2zm-7 5h5v5h-5z"
                  />
                </svg>
              </div>
            </span>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`DatePicker Component Basic DatePicker Component should render the Basic DatePicker Component 1`] = `
<div>
  <div
    class="ant-picker css-dev-only-do-not-override-htwhyh"
  >
    <div
      class="ant-picker-input"
    >
      <input
        autocomplete="off"
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Component Default DatePicker Component should render the Default DatePicker Component 1`] = `
<div>
  <div
    class="ant-picker css-dev-only-do-not-override-htwhyh"
  >
    <div
      class="ant-picker-input"
    >
      <input
        autocomplete="off"
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;

exports[`DatePicker Component should render the DatePicker Component 1`] = `
<div>
  <div
    class="ant-picker css-dev-only-do-not-override-htwhyh"
  >
    <div
      class="ant-picker-input"
    >
      <input
        autocomplete="off"
        placeholder="Select date"
        readonly=""
        size="12"
        title=""
        value=""
      />
      <span
        class="ant-picker-suffix"
      >
        <span
          aria-label="calendar"
          class="anticon anticon-calendar"
          role="img"
        >
          <svg
            aria-hidden="true"
            data-icon="calendar"
            fill="currentColor"
            focusable="false"
            height="1em"
            viewBox="64 64 896 896"
            width="1em"
          >
            <path
              d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
            />
          </svg>
        </span>
      </span>
    </div>
  </div>
</div>
`;
